File: Java/02. Working with Numbers/29. Greatest Common Divisor.java
import java.util.Scanner;

class Main {
    public static void main(String [] args) {
        Scanner userInput = new Scanner(System.in);

        System.out.println("Enter the First number:");
        int number1 = userInput.nextInt();
        
        System.out.println("Enter the Second number:");
        int number2 = userInput.nextInt();
		
        int minimum = number1 < number2 ? number1 : number2;
        int i;

		for( i=minimum; i>=1 ;i-- ){
		    if(( number1 % i == 0 ) && (number2 % i == 0 )){
		        break;
            }
        }
		System.out.println("GCD of " + number1 + " and " + number2 + " is "+ i);
        userInput.close();
    }
}

File: Python/02. Working with Numbers/29. Greatest Common Divisor.py
firstNumber = int(input("Enter the First Number:"))
secondNumber = int(input("Enter the Second Number:"))
number1 = firstNumber
number2 = secondNumber

while (firstNumber != secondNumber):
    if (firstNumber > secondNumber):
        firstNumber -= secondNumber
    else:
        secondNumber -= firstNumber

print("GCD of", number1, "and", number2, "is", firstNumber)